tests/js/realm-tests.js

```diff
@@ -325,6 +332,16 @@ module.exports = BaseTest.extend({
                 realm.create('InvalidObject', {intCol: 1});
             });
         });
+
+        // Only the original constructor should be valid.
+        function WrongCustomObject() {}
+        WrongCustomObject.schema = CustomObject.schema;
```

Comment:
Ok - looks like you are testing that here - WongCustomObject sounds off though, maybe InvalidCustomObject since it isn't wrong but simply not in the schema?